ixmp4/data/db/optimization/utils.py

```diff
@@ -17,37 +19,43 @@ def collect_indexsets_to_check(
     return collection
 
 
-def validate_data(key, data: dict[str, Any], columns: list["Column"]):
+def validate_data(host: base.BaseModel, data: dict[str, Any], columns: list["Column"]):
     data_frame: pd.DataFrame = pd.DataFrame.from_dict(data)
     # TODO for all of the following, we might want to create unique exceptions
     # Could me make both more specific by specifiying missing/extra columns?
     if len(data_frame.columns) < len(columns):
-        raise ValueError(
+        raise host.DataInvalid(
+            f"While handling {host.__str__}: \n"
```

Comment:
Here's to hoping these lines will guide future users in identifying which objects cause the error.

Reply:
`host.__str__` should be a function...